server/services/authorization/canAccessRoom.ts

```diff
@@ -3,11 +3,21 @@ import { Authorization } from '../../sdk';
 import { RoomAccessValidator } from '../../sdk/types/IAuthorization';
 import { canAccessRoomLivechat } from './canAccessRoomLivechat';
 import { canAccessRoomTokenpass } from './canAccessRoomTokenpass';
-import { Subscriptions, Rooms, Settings } from './service';
+import { Subscriptions, Rooms, Settings, TeamMembers } from './service';
 
 const roomAccessValidators: RoomAccessValidator[] = [
 	async function(room, user): Promise<boolean> {
-		if (!room?._id || room.t !== 'c') {
+		if (!room?.teamId) {
+			return false;
+		}
+
+		const team = await TeamMembers.findOne({ userId: user._id, teamId: room.teamId });
```

Comment:
please create a new Model method to execute the find, something like `TeamMembers.findOneByUserIdAndTeamId`.. we should avoid having queries spread in the code.

also since you just want to know if there is a record with `userId` and `teamId`, you can use `{ projection: { _id: 1 } }` to reduce traffic, since you don't need the full record data =)

Reply:
Done!